lib/msf/core/modules/loader/executable.rb

```diff
@@ -88,6 +88,22 @@ def read_module_content(parent_path, type, module_reference_name)
         elog "Unable to load module #{full_path}, unknown module type"
         return ''
       end
+    rescue Errno::ENOENT => e
+      elog("Unable to load module #{full_path}", error: e)
+      # XXX migrate this to a full load_error when we can tell the user why the
+      # module did not load and/or how to resolve it.
+      # load_error(full_path, e)
+
+      # Detect if the module is written in Python or GO, then report it to the user so they
+      # can verify if they have the required runtime language for the module
+      if full_path.end_with?('.py')
+        # print_error('Failed to load external Python module, please ensure you have Python installed on your environment.')
+        puts('[-] Failed to load external GO module, please ensure you have GO installed on your environment.')
+      elsif full_path.end_with?('.go')
+        # print_error('Failed to load external GO module, please ensure you have GO installed on your environment.')
+        puts('[-] Failed to load external GO module, please ensure you have GO installed on your environment.')
+      end
```

Comment:
As above, this seems to be leaky to me. Currently this executable file knows nothing about the types of files it can import, that's handled further down in the stack.

Potentially the validation changes should be implemented lower down within the bridges themselves?

https://github.com/rapid7/metasploit-framework/blob/eefe174e3a4d953877c1cf5b5f194b21e17017b3/lib/msf/core/modules/external/bridge.rb#L219-L224

Depending on how you wanted to implement it, potentially each loader could handle its own verification that the dependencies are loaded correctly and/or raising a specific error message on how to set up the required runtime dependencies if the intiial bridge opening process fails:

https://github.com/rapid7/metasploit-framework/blob/eefe174e3a4d953877c1cf5b5f194b21e17017b3/lib/msf/core/modules/external/bridge.rb#L16-L19